components/gitpod-protocol/go/reconnecting-ws.go

```diff
@@ -205,7 +205,13 @@ func (rc *ReconnectingWebsocket) connect(ctx context.Context) *WebsocketConnecti
 			}
 		}
 
-		if err == websocket.ErrBadHandshake {
+		var statusCode int
+		if resp != nil {
+			statusCode = resp.StatusCode
+		}
+
+		// 200 is bad gateway for ws, we should keep trying
+		if err == websocket.ErrBadHandshake && statusCode != 200 {
```

Comment:
After we do the nil check, we access the `resp` below in 
```
if rc.badHandshakeCount > rc.badHandshakeMax || (http.StatusMultipleChoices <= resp.StatusCode && resp.StatusCode < http.StatusInternalServerError) {
				_ = rc.closeWithError(&ErrBadHandshake{rc.url, rc.reqHeader, resp})
				return nil
			}
```
The access to `resp` happens irrespective if `resp` was nil. Here, we could potentially get a panic from `resp.StatusCode`